React/wbooks/src/WBooks.js

```diff
@@ -1,17 +1,25 @@
-import React from 'react';
-import { Switch, Route } from 'react-router-dom';
+import React, { Component } from 'react';
+import { Switch, Route, withRouter } from 'react-router-dom';
+import { push } from 'react-router-redux';
+import { connect } from 'react-redux';
 import './application.css';
 
 import HomeContainer from './Home/';
 import BookDetail from './BookDetail';
+import LoginContainer from './Login';
+import SignUpContainer from './SignUp';
+import DashboardContainer from './Dashboard';
 
 function WBooks() {
   return (
     <Switch>
       <Route exact path="/" component={HomeContainer} />
       <Route path="/book/:id" component={BookDetail} />
+      <Route path="/login" component={LoginContainer} />
+      <Route path="/signup" component={SignUpContainer} />
+      <Route path="/dashboard" component={DashboardContainer} />
```

Comment:
There's a way to avoid checking if the user is authenticated in each component. You could have a AuthenticatedRoute component that renders the correct route depending on if you pass an isPublic or isPrivate prop. It's long to explain here, so I can show you tomorrow how to do it.

Also `componentWillMount` shouldn't be used any more, because it'll be deprecated in the next React version